wgpu-hal/src/dx12/command.rs

```diff
@@ -767,9 +767,11 @@ impl crate::CommandEncoder<super::Api> for super::CommandEncoder {
                 };
                 self.temp.barriers.push(barrier);
             }
+            profiling::scope!("ID3D12GraphicsCommandList::ResourceBarrier");
```

Comment:
This scope would cover more things than just the next line. If you want this scope, let's do `if !self.temp.barriers.is_empty() { ... }`